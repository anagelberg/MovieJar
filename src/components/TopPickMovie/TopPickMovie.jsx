import './TopPickMovie.scss';
import Rating from '../Rating/Rating';

function TopPickMovie({ movie }) {
    return (
        <div className='top'>
            <div className="top__heading">
                <h4>Top Pick</h4>
            </div>


            <div className="top__body">
                <div className="top__info">
                    <h1>{movie?.title}</h1>
                    <Rating rating={movie?.publicRating} />
                    <p>{movie?.mentalVibe} mentally, {movie?.emotionalVibe} emotionally</p>
                    <p className='top__small'>{movie?.genres}</p>
                    <p>{movie?.description}</p>
                    <p className='top__small'>{movie?.runTime} minutes</p>
                </div>

                <div>
                    <img className='top__img' src={movie?.posterUrl}></img>
                </div>
            </div>

        </div>
    );
}

export default TopPickMovie;