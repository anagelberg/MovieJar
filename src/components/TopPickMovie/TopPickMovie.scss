@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.top {
  @include flex-center($justify-content: space-around);
  @include vertical-margin;
  @include sm-vertical-padding;
  @include sm-side-padding;
  flex-direction: column;
  // gap: $spacing-unit-med;
  width: 100%;
  background-color: $secondary-background;
  border-radius: $border-radius;

  @include tablet {
    flex-direction: row;
  }

  &__img {
    width: 15rem;
    height: auto;
    border-radius: $border-radius;
  }

  &__info {
    display: flex;
    flex-direction: column;
    gap: $spacing-unit-sm;
    max-width: 30rem;
  }

  &__small {
    @include sm-paragraph;
  }
}
