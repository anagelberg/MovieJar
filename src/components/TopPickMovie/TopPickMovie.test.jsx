import TopPickMovie from "./TopPickMovie";
import { render, screen } from "@testing-library/react";


jest.mock('../Rating/Rating', () => (props) => <div data-testid="rating">{props.rating}</div>);

const mockMovie = {
    posterUrl: 'https://image.tmdb.org/t/p/original/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg',
    title: 'Example Movie Title',
    publicRating: 4.5,
    mentalVibe: 'Thought-provoking',
    emotionalVibe: 'Lighthearted',
    genres: 'Action, Adventure',
    description: 'This is a mock description of the movie.',
    runTime: 120
};


describe('TopPickMovie', () => {

    it('renders without crashing', () => {
        render(<TopPickMovie movie={mockMovie} />);
        const movieComponent = screen.getByText(mockMovie.title);
        expect(movieComponent).toBeInTheDocument();
    });

    test('displays the correct movie poster', () => {
        render(<TopPickMovie movie={mockMovie} />);
        const moviePoster = screen.getByRole('img');
        expect(moviePoster).toHaveAttribute('src', mockMovie.posterUrl);
    });

    it('matches snapshot', () => {
        const { container } = render(<TopPickMovie movie={mockMovie} />);
        expect(container.firstChild).toMatchSnapshot();
    });
}); 
