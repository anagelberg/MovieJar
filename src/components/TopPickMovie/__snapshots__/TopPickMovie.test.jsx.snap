// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPickMovie matches snapshot 1`] = `
<div
  class="top"
>
  <div>
    <img
      class="top__img"
      src="https://image.tmdb.org/t/p/original/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
    />
  </div>
  <div
    class="top__info"
  >
    <h2>
      Example Movie Title
    </h2>
    <div
      data-testid="rating"
    >
      4.5
    </div>
    <p>
      Thought-provoking
       mentally, 
      Lighthearted
       emotionally
    </p>
    <p
      class="top__small"
    >
      Action, Adventure
    </p>
    <p>
      This is a mock description of the movie.
    </p>
    <p
      class="top__small"
    >
      120
       minutes
    </p>
  </div>
</div>
`;
